test/views/declaration/CarrierDetailsViewSpec.scala

```diff
@@ -38,482 +37,262 @@ import views.tags.ViewTest
 @ViewTest
 class CarrierDetailsViewSpec extends UnitViewSpec with CommonMessages with Stubs with Injector {
 
-  val form: Form[CarrierDetails] = CarrierDetails.form(STANDARD)
-  private val carrierDetailsPage = instanceOf[carrier_details]
-
-  private def createView(form: Form[CarrierDetails] = form)(implicit request: JourneyRequest[_]): Document =
-    carrierDetailsPage(Mode.Normal, form)(request, messages)
+  val carrierDetailsPage = instanceOf[carrier_details]
+  val allFields = Seq("fullName", "addressLine", "townOrCity", "addressLine", "postCode", "country")
+  val validAddress = Address("Marco Polo", "Test Street", "Leeds", "LS18BN", "England")
+  val emptyAddress = Address("", "", "", "", "")
```

Comment:
Nit: Just figured out that it could be beneficial to extract the test addresses so they can be re-used accross various *Details specs. No need to do it now, unless you want to.

Reply:
If I was more comfortable with the code base and hadn't already spent a couple of days on this then I would. One for the future me thinks ;)